sciter/event.py

```diff
@@ -197,6 +200,54 @@ def _on_script_call(self, f):
                     args = sciter.Value.unpack_from(f.argv, f.argc)
                 else:
                     args = [sciter.Value(f.argv[i]) for i in range(f.argc)]
+
+                if cfg.get('threading'):
+                    # submit this handler to a separate thread
+                    # syntax:
+                    # `root.xcall(name, [args])`
+                    # `root.xcall(name, [args], resolve, reject)`
+                    jsargs = args[0]
+                    jsresolve = args[1] if len(args) == 3 else None
+                    jsreject = args[2] if len(args) == 3 else None
+
+                    def on_script_done(fut):
+                        if fut.cancelled():
+                            return
+
+                        def resolve(val):
+                            if jsresolve is not None:
+                                jsresolve(val)
+                            pass
+
+                        def reject(val):
+                            if jsreject is not None:
+                                jsreject(val)
+                            pass
+
+                        exc = fut.exception()
+                        if exc is None:
+                            # no exception, resolve the promise
+                            rv = fut.result()
+                            resolve(rv)
+                        else:
+                            # there was an exception, reject the promise
+                            # note: reject expects a value, not an error
+                            exc = self.script_exception_handler(fname, exc)
+                            if skip_exception:
+                                resolve(str(exc))
+                            else:
+                                reject(str(exc))
```

Comment:
```py
raise Exception("python exception")
```

Huh! This was a tricky one: `Promise:reject` expects a value, not an error instance. Which is converted to an error later, in `await`.

Otherwise, we are getting some mess like:

```
error:tis: TypeError: cannot read property 'toString' of null
    at <anonymous> (sciter:debug-peer.js:304)
error:tis: Unhandled rejection:null
error:tis: [object]
```

instead of a legit

```
info:tis: async exception,python exception
```

Though I am not entirely sure: [`Promise::Promise()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/Promise) mentions value, but [`Promise::reject()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/reject) mentions an error. Some tests: https://onecompiler.com/javascript/3xv5u752m

cc @AlecHaring